### ISSUE TITLE
TrySaveXML erases existing metadata

### ISSUE THREAD

**Issue by normanb** [OSGeo/gdal#1430]
## Expected behavior and actual behavior.

When asking for stats on a file I would expect the original image metadata to be maintained.

`TileDB::TrySaveXML` in https://github.com/OSGeo/gdal/pull/1402 has the same issue so I will try and fix this there. Creating this issue for tracking purposes.

## Steps to reproduce the problem.

```
wget https://download.osgeo.org/geotiff/samples/spot/chicago/UTM2GTIF.TIF
gdal_translate -OF PNG UTM2GTIF.TIF output.png
cp output.png.aux.xml original.png.aux.xml
gdalinfo -stats output.png
diff original.png.aux.xml output.png.aux.xml
```

I would expect the only difference to be the stats in the new metadata file, the old metadata has been removed.

## Operating system

Ubuntu 18.04 64 bit

## GDAL version and provenance

Current master

**Comment by normanb** [comment]
I put a simple fix in https://github.com/OSGeo/gdal/pull/1402/files#diff-9a223edcf74381629981a9396dcdc1db it is sufficient to check whether the serialized return is null or not.

**Comment by rouault** [comment]
@normanb You can back out your change. I've fixed the issue slightly differently per #1431
